### PR TITLE
test(dash-spv): add multi-wallet integration tests

### PR DESCRIPTION
**Pull request by xdustinface**
Adds `tests_multi_wallet.rs` plus extensions to `tests_basic.rs` and `tests_restart.rs` covering per-wallet sync behavior:

- Two wallets in one `WalletManager` sync independently.
- Runtime wallet-add with `birth_height` before, at, and beyond tip, asserting existing wallets' `synced_height` and `last_processed_height` never regress.
- Deep historical rescan triggered by a `birth_height = 0` runtime add.
- A block funding two wallets is attributed correctly to each after both are added at runtime.
- Runtime add against a `Syncing`-state `FiltersManager`.
- Runtime-add rescan composed with a live tip advance.
- Restart of a fully-synced client emits no rescan, then runtime-adding a second wallet only catches up the new wallet.

Based on: 
- #692 
- #693 
- #694 

## Summary by CodeRabbit

* **Tests**
  * Expanded test coverage for multi-wallet SPV synchronization scenarios, including wallet isolation and edge case handling during syncing
  * Introduced test infrastructure improvements with helper utilities for wallet setup and synchronization verification
  * Added new test cases covering runtime wallet additions and complex synchronization behavior across multiple wallet configurations

### CODE REVIEW

File: dash-spv/tests/dashd_sync/helpers.rs [hunk 1]
```diff
@@ -86,6 +86,50 @@ pub(super) async fn get_spendable_balance(
...
+    let deadline = tokio::time::Instant::now() + Duration::from_secs(30);
+    loop {
+        let recv = tokio::time::timeout_at(deadline, events.recv()).await;
+        match recv {
+            Err(_) => {
+                panic!("wallet did not reach synced_height {} within 30s, got {}", target, synced)
+            }
+            Ok(Err(_)) => {
+                panic!("wallet event channel error before reaching synced_height {}", target);
+            }
```
[the remark this answers is not in the document]
Comment (xdustinface) [reply dash-spv/tests/dashd_sync/helpers.rs:128]: We dont have lagged event recovery its intentionally.

**Review by QuantumExplorer — APPROVED** [review]
Rubber stamping
